Code quality

gdestuynder opened the pull request.

andrewkrug committed:
various pep8 fixes

andrewkrug committed:
various_pep8 fixes

gdestuynder commented on src/flask_pyoidc/flask_pyoidc.py:
```diff
@@ -1,50 +1,135 @@
+        return last + refresh
+
+    def authenticated(self):
+        if self.flask_session.get('id_token_jwt'):
```
im guessing ideally this should call oic to check that the token is valid instead

andrewkrug replied:
This should be easy-ish?

andrewkrug replied:
would this more or less be sharing code with https://github.com/mozilla-iam/Flask-pyoidc/blob/code_quality/src/flask_pyoidc/flask_pyoidc.py#L188

andrewkrug replied:
If that's the case I can move it out to module scope or something for token validation.

gdestuynder commented on src/flask_pyoidc/flask_pyoidc.py:
```diff
@@ -1,50 +1,135 @@
+        # dynamically add the flask redirect uri to the client info
         with self.app.app_context():
-            self.client_registration_info['redirect_uris'] = url_for('redirect_uri')
+            self.client_registration_info['redirect_uris']\
```
nit ' \\' vs '\\' earlier

gdestuynder commented on src/flask_pyoidc/flask_pyoidc.py:
```diff
@@ -1,50 +1,135 @@
 
-        if client_registration_info and 'client_id' in client_registration_info:
+        # if non-discovery client add the provided info from the constructor
+        if client_registration_info and 'client_id'\
```
same

gdestuynder commented on src/flask_pyoidc/flask_pyoidc.py:
```diff
@@ -102,14 +217,21 @@ def _handle_authentication_response(self):
             'code': authn_resp['code'],
-            'redirect_uri': self.client.registration_response['redirect_uris'][0],
+            'redirect_uri':
+                self.client.registration_response['redirect_uris'][0]
```
nit: should this be passed to the function instead? ie allow to pick which redirect uri instead of the first one, if needed by users of the library

andrewkrug replied:
I really actually dislike all of the way this handles redirect urls.  As a user I'd like to be able to set this in the config and use an object to know what redirect URL I'm asking for.

gdestuynder replied:
"yes" ;-)

gdestuynder commented on src/flask_pyoidc/flask_pyoidc.py:
```diff
@@ -121,52 +243,86 @@ def _handle_authentication_response(self):
             if current_app.config.get('SESSION_PERMANENT'):
                 flask.session.permanent = True
-                flask.session.permanent_session_lifetime = id_token.get('exp')-time.time()
+                flask.session.permanent_session_lifetime\
```
' \\'

gdestuynder commented on src/flask_pyoidc/flask_pyoidc.py:
```diff
@@ -121,52 +243,86 @@ def _handle_authentication_response(self):
 
-        return 'Something went wrong with the authentication, please try to login again.'
+        message = "Something went wrong with the authentication, \
+        please try to login again."
```
nit this adds leading spaces to the msg

gdestuynder commented on src/flask_pyoidc/flask_pyoidc.py:
```diff
@@ -198,7 +367,8 @@ def oidc_logout(self, view_func):
             if 'state' in flask.request.args:
                 # returning redirect from provider
-                assert flask.request.args['state'] == flask.session.pop('end_session_state')
+                assert flask.request.args['state']\
```
' \\'

gdestuynder commented on tests/test_flask_pyoidc.py:
```diff
@@ -12,15 +12,180 @@
+            flask.session['access_token'] = 'test token'
+            flask.session['id_token'] = id_token.to_dict()
+            flask.session['id_token_jwt'] = id_token.to_jwt()
+            flask.session['last_authenticated'] = time.time() + 100
```
just to follow other changes s/time.time()/now/

gdestuynder committed:
Nit fixes

gdestuynder committed:
Comment to clarify how def authenticated() really works
Since this function is just a prettified wrapper to check if the
session is still a valid flask session.
